src/expr.rs

```diff
@@ -245,8 +245,8 @@ fn format_expr(expr: &ast::Expr,
         ast::ExprKind::InlineAsm(..) => {
             wrap_str(context.snippet(expr.span), context.config.max_width, shape)
         }
-        // FIXME(#1537)
-        ast::ExprKind::Catch(..) => unimplemented!(),
+        // 6 = `catch `
+        ast::ExprKind::Catch(ref block) => block.rewrite(&context, try_opt!(shape.sub_width(6))),
```

Comment:
Thanks!
It would be nice if rustfmt could handle single line case inside `impl Rewrite for ast::Block` along with `unsafe` ([here](https://github.com/topecongiro/rustfmt/blob/libsyntax/src/expr.rs#L663-L672)). However, as far as I know, `ast::Block` does not carry information whether it is `catch` block or not. Hence we need to handle single line case for `catch` before calling `block.rewrite`[here](https://github.com/topecongiro/rustfmt/blob/libsyntax/src/expr.rs#L249). Is this correct?